xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/PropertyClass.java

```diff
@@ -293,6 +293,12 @@ public void displayCustom(StringBuffer buffer, String fieldName, String prefix,
                 ScriptContext.ENGINE_SCOPE);
             scontext.setAttribute("object", new com.xpn.xwiki.api.Object(object, context), ScriptContext.ENGINE_SCOPE);
             scontext.setAttribute("type", type, ScriptContext.ENGINE_SCOPE);
+            /* This is a text alternative fallback to explain what the input is about. 
```

Comment:
Does having an `aria-label` only apply to `PropertyClass` and `StringClass` only?
If we provide a default generic text alternative, it is probably better if we try to cover all cases.

Reply:
> Before reaching this solution, I tried to provide a generic fix for all Property types. However, the `displayEdit` function is overridden for a lot of object classes. This solution only fix String properties, but a similar solution could be used to provide text alternatives for the inputs of other property types.

Since there's no straightforward way to generalize this without updating all of the classes implementations for `displayEdit`, I don't think I want to merge it all at once.

These changes have a wide impact, so I'd rather keep them as minimal as necessary to avoid breaking things. See for example the number of test updates needed just for what was done until now.

Moreover it doesn't seem to me like it's needed for other property types. This might already be too generalized, in practice I only *need* this change to the displayEdit function for https://github.com/xwiki/xwiki-platform/blob/e5357c39ba2abd648cfbb0ac1b650a23df7d8d7e/xwiki-platform-core/xwiki-platform-release/xwiki-platform-release-ui/src/main/resources/ReleaseCode/ReleaseSheet.xml#L53-L63 . I could probably just have hardcoded a text alternative in this form sheet.

WDYT?